From 2b46b7a8e9fdfdd8384a6a33b8fea5f01b7a7a68 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 02 Aug 2022 16:57:25 +0300
Subject: [PATCH] [SW-98239] erom: recovery: remove redundant version

The version related defines and arrays are used for dynamic security
init.
Those are used incorrectly because they do not contain the correct
version information. Additionally, starting from Gaudi2 the version
information is no longer provided in SRAM and therefore the
preboot_security_dynamic_vers_init() function should not be called.
Remove these redundant pieces of code.

Change-Id: I46fe539967d4731ba7d4b9a8c6d971edb0771869
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index 67cb4a9..70542c7 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -17,14 +17,6 @@
 
 #include <uart.h>
 
-#define BTL_VER_PREFIX  "BTL version "
-#define FUSE_VER_PREFIX "Fuse version "
-
-static char preboot_vers[][VERSION_MAX_LEN] = {
-	[PREBOOT_BTL_VER] = { BTL_VER_PREFIX },
-	[PREBOOT_EFUSE_VER] = { FUSE_VER_PREFIX },
-};
-
 static struct cmd_module_ops cmd_ops[] = {
 	{	/* PCI module */
 		.setup = module_pci_setup,
@@ -89,8 +81,6 @@
 
 void erom_main_function(void)
 {
-	u64 ver_off = (uintptr_t)preboot_vers;
-
 	device_init();
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
@@ -126,7 +116,6 @@
 
 	/* Prepare boot FW versions and eFuse data for host */
 	preboot_version();
-	preboot_security_dynamic_vers_init(ver_off, sizeof(preboot_vers));
 
 	preboot_security_sp_sram_init();
 	preboot_dev_enable_caps();
